src/network/connection.cpp

```diff
@@ -691,15 +665,19 @@ void Connection::update() {
 
 			for (int i = 0; i < (int)sensors.size(); i++) {
 				if (sensorInfoPacket.sensorId == sensors[i]->getSensorId()) {
-					m_AckedSensorState[i] = sensorInfoPacket.sensorState;
 					if (len < 12) {
-						m_AckedSensorCalibration[i]
-							= sensors[i]->hasCompletedRestCalibration();
-						m_AckedSensorConfigData[i] = sensors[i]->getSensorConfigData();
+						sensors[i]->signalAckedStateUpdate(
+							sensorInfoPacket.sensorState,
+							sensors[i]->hasCompletedRestCalibration(),
+							sensors[i]->getSensorConfigData()
+						);
 						break;
 					}
-					m_AckedSensorCalibration[i]
-						= sensorInfoPacket.hasCompletedRestCalibration;
+					sensors[i]->signalAckedStateUpdate(
+						sensorInfoPacket.sensorState,
+						sensorInfoPacket.hasCompletedRestCalibration,
+						sensorInfoPacket.sensorConfigData
```

Comment:
It seems like this line has been forgotten in the code before.
But i'm leads to that what we want. How does the server request a change as example for the magnetometer?
I think i need to test that.

Reply:
testet